resources/qml/WelcomePages/AddLocalPrinterScrollView.qml

```diff
@@ -51,103 +58,164 @@ ScrollView
-                    height: UM.Theme.getSize("standard_arrow").height
-                    sourceSize.width: width
-                    sourceSize.height: height
-                    color: UM.Theme.getColor("text")
-                    source: base.currentSection == section ? UM.Theme.getIcon("arrow_bottom") : UM.Theme.getIcon("arrow_right")
+                    id: machineDefinitionsModel
+                    filter: { "visible": true }
+                    sectionProperty: "category"
+                    preferredSectionValue: preferredCategory
                 }
 
-                Label
+                section.property: "section"
+                section.delegate: sectionHeader
+                delegate: machineButton
+            }
+
+            Component
+            {
+                id: sectionHeader
+
+                Button
                 {
-                    id: label
-                    anchors.left: arrow.right
-                    anchors.leftMargin: UM.Theme.getSize("default_margin").width
-                    verticalAlignment: Text.AlignVCenter
-                    text: button.text
-                    font.bold: true
-                    renderType: Text.NativeRendering
+                    id: button
+                    width: ListView.view.width
+                    height: UM.Theme.getSize("action_button").height
+                    text: section
+
+                    property bool isActive: base.currentSection == section
+
+                    background: Rectangle
+                    {
+                        anchors.fill: parent
+                        color: isActive ? UM.Theme.getColor("setting_control_highlight") : "transparent"
+                    }
+
+                    contentItem: Item
+                    {
+                        width: childrenRect.width
+                        height: UM.Theme.getSize("action_button").height
+
+                        UM.RecolorImage
+                        {
+                            id: arrow
+                            anchors.left: parent.left
+                            width: UM.Theme.getSize("standard_arrow").width
+                            height: UM.Theme.getSize("standard_arrow").height
+                            sourceSize.width: width
+                            sourceSize.height: height
+                            color: UM.Theme.getColor("text")
+                            source: base.currentSection == section ? UM.Theme.getIcon("arrow_bottom") : UM.Theme.getIcon("arrow_right")
+                        }
+
+                        Label
+                        {
+                            id: label
+                            anchors.left: arrow.right
+                            anchors.leftMargin: UM.Theme.getSize("default_margin").width
+                            verticalAlignment: Text.AlignVCenter
+                            text: button.text
+                            font.bold: true
```

Comment:
We should just get a font from the theme (because this could mean it gets a different font type)